cuda_core/tests/test_program.py

```diff
@@ -9,10 +9,189 @@
 from cuda.core.experimental import _linker
 from cuda.core.experimental._module import Kernel, ObjectCode
 from cuda.core.experimental._program import Program, ProgramOptions
+from cuda.core.experimental._utils.cuda_utils import driver, handle_return
 
+cuda_driver_version = handle_return(driver.cuDriverGetVersion())
 is_culink_backend = _linker._decide_nvjitlink_or_driver()
 
 
+def _is_nvvm_available():
+    """Check if NVVM is available."""
+    try:
+        from cuda.core.experimental._program import _get_nvvm_module
+
+        _get_nvvm_module()
+        return True
+    except RuntimeError:
+        return False
+
+
+nvvm_available = pytest.mark.skipif(
+    not _is_nvvm_available(), reason="NVVM not available (libNVVM not found or cuda-bindings < 12.9.0)"
+)
+
+try:
+    from cuda.core.experimental._utils.cuda_utils import driver, handle_return
+
+    _cuda_driver_version = handle_return(driver.cuDriverGetVersion())
+except Exception:
+    _cuda_driver_version = 0
+
+_libnvvm_version = None
+_libnvvm_version_attempted = False
+
+precheck_nvvm_ir = """target triple = "nvptx64-unknown-cuda"
+target datalayout = "e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-i128:128:128-f32:32:32-f64:64:64-v16:16:16-v32:32:32-v64:64:64-v128:128:128-n16:32:64"
+
+define void @dummy_kernel() {
+  entry:
+    ret void
+}
+
+!nvvm.annotations = !{{!0}}
+!0 = !{{void (i32*)* @simple, !"kernel", i32 1}}
+
+!nvvmir.version = !{{!1}}
+!1 = !{{i32 {major}, i32 0, i32 {debug_major}, i32 0}}
+"""  # noqa: E501
+
+
+def _get_libnvvm_version_for_tests():
+    """
+    Detect libNVVM version by compiling dummy IR and analyzing the PTX output.
+
+    Workaround for the lack of direct libNVVM version API (nvbugs 5312315).
+    The approach:
+    - Compile a small dummy NVVM IR to PTX
+    - Use PTX version analysis APIs if available to infer libNVVM version
+    - Cache the result for future use
+    """
+    global _libnvvm_version, _libnvvm_version_attempted
+
+    if _libnvvm_version_attempted:
+        return _libnvvm_version
+
+    _libnvvm_version_attempted = True
+
+    try:
+        from cuda.core.experimental._program import _get_nvvm_module
+
+        nvvm = _get_nvvm_module()
+
+        try:
+            from cuda.bindings.utils import get_minimal_required_cuda_ver_from_ptx_ver, get_ptx_ver
+        except ImportError:
+            _libnvvm_version = None
+            return _libnvvm_version
+
+        program = nvvm.create_program()
+        try:
+            precheck_ir_bytes = precheck_nvvm_ir.encode("utf-8")
```

Comment:
Please make sure the changes are tested locally.
```suggestion
            major, minor, debug_major, debug_minor = nvvm.ir_version()
            precheck_nvvm_ir.format(major=major, minor=minor, debug_major=debug_major, debug_minor=debug_minor)
            precheck_ir_bytes = precheck_nvvm_ir.encode("utf-8")
```